code/iaas/logic/src/main/java/io/cattle/platform/process/containerevent/ContainerEventCreate.java

```diff
@@ -61,76 +68,109 @@
     AccountDao accountDao;
 
     @Inject
+    InstanceDao instanceDao;
+
+    @Inject
     LockManager lockManager;
 
     @SuppressWarnings("unchecked")
     @Override
     public HandlerResult handle(ProcessState state, ProcessInstance process) {
-        if ( !MANAGE_NONRANCHER_CONTAINERS.get() ) {
+        if (!MANAGE_NONRANCHER_CONTAINERS.get()) {
             return null;
         }
 
-        ContainerEvent event = (ContainerEvent)state.getResource();
-        Map<String, Object> inspect = (Map<String, Object>)DataUtils.getFields(event).get(FIELD_DOCKER_INSPECT);
+        final ContainerEvent event = (ContainerEvent)state.getResource();
 
-        String rancherUuid = getRancherUuidLabel(inspect);
-        if ( rancherUuid != null ) {
-            return null;
-        }
+        HandlerResult result = lockManager.lock(new ContainerEventInstanceLock(event.getAccountId(), event.getExternalId()), new LockCallback<HandlerResult>() {
```

Comment:
In the ContainerEventInstanceLock I would probably hash the external id when constructing the lock key.  You can even just take the hashCode().  The reason is that this information could be sent from external so could have bad values in it.  For example, a "/" in a lock id can cause problems on ZooKeeper.